cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSISink.java

```diff
@@ -403,22 +421,51 @@ private Status processNewBatches() throws EventDeliveryException {
                 LOGGER.error("Runtime error (" + e.getMessage() + ")");
             } // catch
 
-            // parse the event and accumulate it
-            try {
+            if (event instanceof com.telefonica.iot.cygnus.interceptors.NGSIEvent) {
+                NotifyContextRequest originalNCR =
+                        ((com.telefonica.iot.cygnus.interceptors.NGSIEvent) event).getOriginalNCR();
+                NotifyContextRequest mappedNCR =
+                        ((com.telefonica.iot.cygnus.interceptors.NGSIEvent) event).getMappedNCR();
+
+                // Accumulate the event
+                try {
+                    accumulator.accumulate(event.getHeaders(), originalNCR, mappedNCR);
+                    numProcessedEvents++;
+                } catch (Exception e) {
+                    LOGGER.error("There was some problem when accumulating the notified context element. "
+                            + "Details: " + e.getMessage());
+                } // try catch
+            } else {
+                // 'TODO': to be removed
                 LOGGER.debug("Event got from the channel (id=" + event.hashCode() + ", headers="
                         + event.getHeaders().toString() + ", bodyLength=" + event.getBody().length + ")");
-                NotifyContextRequest notification = parseEventBody(event);
-                accumulator.accumulate(event.getHeaders(), notification);
-                numProcessedEvents++;
-            } catch (Exception e) {
-                LOGGER.debug("There was some problem when parsing the notifed context element. Details="
-                        + e.getMessage());
-            } // try catch
+
+                // Parse the event
+                NotifyContextRequest notification;
+                
+                try {
+                    notification = parseEventBody(event);
+                } catch (Exception e) {
+                    LOGGER.error("There was some problem when parsing the notifed context element. Details: "
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1198/commits/3f5e44ab8c66588e5fa37a684e128ebd50852f79